### PR TITLE
Fixes an issue with Hivebot bombers

### PR DESCRIPTION
**Pull request by fernerr**
Basically if two exploded in close proximity to each other it could lead to infinite recursion, crashing the server.

### CODE REVIEW

**Comment by fernerr** [comment]
!bugfix

**Comment by fernerr** [comment]
!review

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 1]
```diff
@@ -75,24 +75,27 @@
...
 	wander = 0
-
-	playsound(src.loc, 'sound/items/countdown.ogg', 125, 1)
-	spawn(20)
-		fragem(src,10,30,2,3,5,1,0)
-		src.gib()
+	if(!has_exploded)
+		playsound(src.loc, 'sound/items/countdown.ogg', 125, 1)
+		has_exploded = 1
+		spawn(20)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: should be a timer imho but i understand if its out of scope

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 2]
```diff
@@ -75,24 +75,27 @@
...
 	stop_automated_movement = 1
 	wander = 0
-
-	playsound(src.loc, 'sound/items/countdown.ogg', 125, 1)
-	spawn(20)
-		fragem(src,10,30,2,3,5,1,0)
-		src.gib()
+	if(!has_exploded)
+		playsound(src.loc, 'sound/items/countdown.ogg', 125, 1)
+		has_exploded = 1
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: true/false pls

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 3]
```diff
@@ -75,24 +75,27 @@
 	icon_state = "hivebotbomber"
 	attacktext = "bumped"
 	move_to_delay = 8
+	var/has_exploded
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]:
```suggestion
	var/has_exploded = FALSE
```

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm (hunk 2 above)
Comment (skull132) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]:
```suggestion
		has_exploded = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 4]
```diff
@@ -75,24 +75,27 @@
...
+			fragem(src,10,30,2,3,5,1,0)
+			src.gib()
 
 /mob/living/simple_animal/hostile/hivebot/bomber/bullet_act(var/obj/item/projectile/Proj)
 	if(istype(Proj, /obj/item/projectile/bullet/pistol/hivebotspike) || istype(Proj, /obj/item/projectile/beam/hivebot))
 		Proj.no_attack_log = 1
 		return PROJECTILE_CONTINUE
-	else
+	else if(!has_exploded)
+		has_exploded = 1
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]:
```suggestion
		has_exploded = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm (hunk 1 above)
Comment (fernerr) [reply code/modules/mob/living/simple_animal/hostile/hivebot.dm]: Replaced it with a callback timer

**Pull request merged**